clients/js/src/index.ts

```diff
@@ -39,3 +39,52 @@ export class CacReader {
         return deepMerge(targetConfig, ...requiredOverrides);
     }
 }
+
+export class ExperimentReader {
+    experiments: Experiments;
+
+    constructor(experiments: Experiments) {
+        this.experiments = experiments;
+    }
+
+    public getApplicableVariant(data: IObject, toss: number): Array<String> {
+        const experiments = this.getSatisfiedExperiments(data);
+        const variants = [];
+        for (const exp of experiments) {
+            const v = this.decideVariant(exp.traffic_percentage, exp.variants, toss);
+            if (v) {
+                variants.push(v.id)
+            }
+        }
+        return variants;
+    };
+
+    getSatisfiedExperiments(data: IObject): Experiments {
+        return this.experiments.filter(exp => jsonLogic.apply(exp.context, data));
+    };
+
+    // decide which variant to return among all applicable experiments
+    decideVariant(
+        traffic: number,
+        applicable_variants: Variants,
+        toss: number,
+    ): Variant | undefined {
+        if (!Number.isInteger(toss)) {
+            return undefined;
+        }
+        if (toss < 0) {
```

Comment:
can we check that toss is an integer too?

Reply:
have added the check [here](https://github.com/juspay/superposition/pull/26/files/f0b8b8802e60d6325d131caf71de0107c37b8424#diff-8392fd7ebcc540bad7b0c0d66740da3daf807d32253418c975825c96fa881bb5R72)